packages/jaeger-ui/src/utils/tracking/index.tsx

```diff
@@ -12,71 +12,47 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import _get from 'lodash/get';
-import queryString from 'query-string';
-import ReactGA from 'react-ga';
-import Raven, { RavenOptions, RavenTransportOptions } from 'raven-js';
-
-import convRavenToGa from './conv-raven-to-ga';
+import { TNil, IWebAnalytics } from '../../types';
+import GA from './ga';
 import getConfig from '../config/get-config';
-import { TNil } from '../../types';
 
-const EVENT_LENGTHS = {
-  action: 499,
-  category: 149,
-  label: 499,
-};
+const TrackingImplementation = () => {
+  const config = getConfig();
 
-// Util so "0" and "false" become false
-const isTruish = (value?: string | string[]) => Boolean(value) && value !== '0' && value !== 'false';
+  const GenericWebAnalytics: IWebAnalytics = {
+    init: () => {},
+    trackPageView: () => {},
+    trackError: () => {},
+    trackEvent: () => {},
+    context: null,
+    isEnabled: () => false,
+  };
 
-const isProd = process.env.NODE_ENV === 'production';
-const isDev = process.env.NODE_ENV === 'development';
-const isTest = process.env.NODE_ENV === 'test';
+  let webAnalytics = GenericWebAnalytics;
 
-// In test mode if development and envvar REACT_APP_GA_DEBUG is true-ish
-const isDebugMode =
-  (isDev && isTruish(process.env.REACT_APP_GA_DEBUG)) ||
-  isTruish(queryString.parse(_get(window, 'location.search'))['ga-debug']);
+  if (config.tracking && config.tracking.customWebAnalytics) {
+    webAnalytics = config.tracking.customWebAnalytics(config) as IWebAnalytics;
+  } else if (config.tracking && config.tracking.gaID) {
+    webAnalytics = new GA(config);
+  }
```

Comment:
Another option instead of forced priority order is to allow enabling both GA and custom WA via a wrapper implementation that would dispatch events to both. But I don't feel strongly about it, the priority order seems fine.